pkg/cloudprovider/vsphere/loadbalancer/README.md

```diff
@@ -0,0 +1,111 @@
+# NSX-T Load Balancer Controller
+
+*Kubernetes load balancer support using NSX-T for the `vSphere` cloud controller
+manager.*
+
+This package enriches the cloud provider interface by implementing the load
+balancing API of the cloud controller for an NSX-T environment.
+
+There are two different modes:
+
+- the *managed* mode manages the load balancer service. The NSX-T load balancer
+  service is only created if it is required.
+  This saves resources if no kubernetes service of type `LoadBalancer` is
+  actually used.
+- the *unmanaged* mode is used if the configuration specifies a load balancer
+  service id. Here only the virtual servers are managed.
+
+The basic assumption is that all nodes are bound to a logical tier1 router.
```

Comment:
does this mean in `managed` mode, the Logical Router ID will be inferred by looking at the router one of the nodes is connected to?
do we have validation on this assumption `all nodes are bound to a logical tier1 router`?

Reply:
I think logical router ID is required in the config

Reply:
not sure about nodes being bound to tier 1 router though

Reply:
Please correct me, if I'm wrong, but as far as we know there can be only one LBService per logical router/tier1 gateway. The same way there can be only one router configured for an LBService. 
This would mean to manage multiple VIPs and LBServices and Virtual servers dependening on the router used for a node, if the nodes are connected to different routers.
I ' don't know whether this is a useful scenario. It could be useful for HA purposes, but as far as I understood from some workshop with VMware, the edge clusters provide an HA setup for the NSX-T elements. So, there should be no need to add nodes to multiple segments handled by different routers.

This would significantly complicate the complete LB handling and enforce to use some DNS based load balancing. I think we don't need this scenario.

Reply:
sorry for my late response.
@mandelsoft
No, even one node is connected with only one router, how could we ensure all nodes connect to the same router? also could you please clarify if the logical router ID is required in config or not?

Reply:
nvm.  I'm ok with this as the starting point.
```
- the *managed* mode manages the load balancer service, also. Here the tier1
  gateway must be specified used, which is used for the segments the cluster
  nodes are connected to. The NSX-T load balancer service is only created if it
  is required. This saves resources if no kubernetes service of type
  `LoadBalancer` is actually used.
```